substrate/client/cli/src/config.rs

```diff
@@ -638,6 +644,14 @@ pub trait CliConfiguration<DCV: DefaultConfigurationValues = ()>: Sized {
 
 		logger.init()?;
 
+		if config.role.is_authority() && config.network.public_addresses.is_empty() {
```

Comment:
Did you check parachain collators(is_authorithy returns true) won't get this warning?

Reply:
https://github.com/paritytech/polkadot-sdk/issues/6280